### ISSUE TITLE
Classification based on target probability distribution

### ISSUE THREAD

**Issue by invemichele** [jhenin/Methods-for-enhanced-sampling-and-free-energy-calculations#4]
First of all, I really liked the paper and the idea of making it a live review, thanks for this contribution!

I want to propose another classification for enhanced sampling methods, that would be complementary to the main one adopted by the paper (Fig. 1). The idea is to look at the probability distribution sampled at convergence by the methods, instead of focusing on how they sample it. In [this paper](https://journals.aps.org/prx/abstract/10.1103/PhysRevX.10.041034) we call it _target distribution_ (TD), and is a probability distribution over the whole configuration space (thus has a slightly different meaning than the one in the review).
This target distribution should be easy to sample and fully cover the Boltzmann distribution of interest.

Here is a first draft of how this TD classification might look like:
![Target distribution classification scheme](https://user-images.githubusercontent.com/14904699/154850665-e4689f44-d8e9-4ed3-a826-2725507661d5.png)
An enhanced sampling method might support one or more target distributions, and vice versa the same TD can be sampled by more than one method. This classification does not focus on the specific target distribution (this can also be interesting, but there are a lot of them), but rather on the general ways used to define them.

**Why is it interesting?** Thinking about the target distribution helps comparing methods, and understanding what they can and cannot achieve. For instance, if I choose a temperature-expanded TD to study an entropy-driven transition (e.g. first order phase transition), I will not obtain great results, regardless of how efficient the method I use is. The mistake is in the choice of the target, not in the details of the enhanced sampling algorithm.
This classification can also help navigating the various hybrid methods.

It would be a small addition to the paper, but hopefully useful. If you think it might be of interest, I can elaborate further on the idea. You can also find more details in the paper I linked above.

**Comment by valsson** [comment]
Hello Michele, 

Thanks for the comment! This is certainly an interesting viewpoint that is worth adding to the review. 

For the "Forced Marginal Distribution" branch, I can think of methods such as metadynamics, VES, OPES, and other CV-based methods. Do you have some examples of methods that are not CV-based methods? Off the top of my mind, I cannot think of any that are not CV-based. 

Also, could you maybe give a few examples of methods for each final branch? That would help us in how to think about this classification. 

Best regards,
Omar

**Comment by mrshirts** [comment]
@invemichele could you describe a bit more about targeting a non-equilibrium distribution? I'm not sure that's well-defined, as nonequilibrium implies a history-dependent family of distributions.

**Comment by invemichele** [comment]
@valsson yes, most methods in that category are CV-based methods (adaptive bias/forces methods, adiabatic methods). The other example I had in mind are multicanonical methods (e.g. see [link](https://pubs.acs.org/doi/10.1021/jp962142e)) that force a flat potential-energy marginal. There are also other methods, but to my knowledge are only used with Monte Carlo schemes (e.g. nested sampling).

Expanded (or generalized ensembles) TD methods have as target a sum of overlapping distributions. Examples are replica exchange, simulated tempering, integrated tempering, thermodynamic integration, multiple windows umbrella sampling, etc. Some of them run independent simulations and calculate the proper weighting at post processing, other do it on-the-fly. 

The main difference in the two categories is how the target distribution is defined, not what specific target distribution is sampled. For example, in the limit of a large number of narrow umbrella windows, one samples a uniform distribution along the CV, and conversely one can define a custom marginal distribution over the energy that leads to the same TD of a parallel tempering run.
There are general characteristics shared by most of the methods in the same category. Methods that define the TD by its marginal, must store an estimate of this marginal (or of the unbiased marginal, or of its derivatives). Thus they deal with histograms, kernel density estimation, basis set expansion, or similar tools. Expanded TD methods generally don't, but must deal with the problem of spacing the intermediate states to have good overlap, and typically use estimators such as MBAR, WHAM, etc.

@mrshirts you are right, I did not pay much attention to the out-of-equilibrium methods. If what they sample is out of equilibrium, then it's not a well-defined distribution. Probably then the classification should distinguish between methods that aim at sampling a target distribution and methods that don't. 
(Incidentally, the TD is well-defined but unknown beforehand, which is why nontrivial sampling methods are needed)

</br>
Anyway, I am sure the classification I proposed can be improved and probably I am missing or misrepresenting some methods. For example, the final division in uniform and non-uniform is motivated by the fact that most methods sample the uniform type, but maybe it is not very useful. My main point is that focusing on what the different methods sample at convergence can be helpful in navigating the sea of enhanced sampling.
